kidneys_misc.module

```diff
@@ -4,6 +4,11 @@
  * Implements hook_form_alter
  */
 function kidneys_misc_form_alter(&$form, &$form_state, $form_id) {
+  // Add custom validation for entityforms.
+  if (isset($form["#entity_type"]) && $form["#entity_type"] == 'entityform' && !(path_is_admin(current_path()))) {
+    $form['#validate'][] = 'kidneys_misc_trial_validate';
```

Comment:
Unfortunately, I think the only way to do this would be to introduce additional config in the code and check for the presence of an id that they specify somewhere. Per your comment below, I like the idea of adding handling for default cases to validation function. Just because I would prefer to limit the coding changes to one place to make sure they don't miss something.